tests/test-regression.C

```diff
@@ -0,0 +1,74 @@
+// Copyright(c)'1994-2025 by The Givaro group
+// This file is part of Givaro.
+// BB: adapted,enhanced from examples/FiniteFields/ff-arith.C
+// Givaro is governed by the CeCILL-B license under French law
+// and abiding by the rules of distribution of free software.
+// see the COPYRIGHT file for more details.
+
+#include <iostream>
+
+#include <givaro/givrational.h>
+using namespace Givaro;
+
+
+bool testRationalDenom() {
+
+    bool pass = true;
+
+    QField<Rational> QQ;
+
+    Rational a(-1,1), b(1,-1), c;
+    std::clog << "-1: " << a << std::endl;
+    std::clog << "-1: " << b << std::endl;
+    pass &= (a == QQ.mOne);
```

Comment:
Warning: &= is bitwise and + assignment. With bool, it would be safer to use pass = pass && (<condition>);

Reply:
I've pushed a fix to this remark